spec/factories/trainees.rb

```diff
@@ -148,45 +148,60 @@
       training_route { TRAINING_ROUTE_ENUMS[:school_direct_salaried] }
     end
 
+    trait :with_trn do
+      trn { SecureRandom.uuid }
+      dttp_id { SecureRandom.uuid }
+      submitted_for_trn_at { Time.zone.now }
+    end
+
     trait :draft do
       state { "draft" }
     end
 
     trait :submitted_for_trn do
-      state { "submitted_for_trn" }
-      submitted_for_trn_at { Time.zone.now }
+      completed
       dttp_id { SecureRandom.uuid }
+      submitted_for_trn_at { Time.zone.now }
+      state { "submitted_for_trn" }
     end
 
     trait :trn_received do
-      submitted_for_trn
+      completed
+      with_trn
       state { "trn_received" }
-      dttp_id { SecureRandom.uuid }
     end
 
     trait :recommended_for_award do
-      state { "recommended_for_award" }
+      completed
+      with_trn
       recommended_for_award_at { Time.zone.now }
+      state { "recommended_for_award" }
     end
 
     trait :withdrawn do
-      state { "withdrawn" }
+      completed
+      with_trn
       withdraw_date { Faker::Date.in_date_period }
+      state { "withdrawn" }
     end
 
     trait :deferred do
-      submitted_for_trn
-      state { "deferred" }
+      completed
+      with_trn
       defer_date { Faker::Date.in_date_period }
+      state { "deferred" }
     end
 
     trait :reinstated do
-      state { "trn_received" }
+      completed
       defer_date { Faker::Date.in_date_period }
       reinstate_date { Faker::Date.in_date_period }
+      state { "trn_received" }
     end
 
     trait :awarded do
+      completed
```

Comment:
["draft", "submitted_for_trn", "trn_received", "recommended_for_award", "awarded"]

You probably wanna chain these, as it literal movement.

Reply:
Good idea - updated now so that each trait uses the state before it and then makes the necessary changes.